nodecontrol/algodControl.go

```diff
@@ -283,8 +288,18 @@ func (nc NodeController) GetAlgodPath() string {
 
 // Clone creates a new DataDir based on the controller's DataDir; if copyLedger is true, we'll clone the ledger.sqlite file
 func (nc NodeController) Clone(targetDir string, copyLedger bool) (err error) {
-	os.RemoveAll(targetDir)
-	err = os.Mkdir(targetDir, 0700)
+	err = os.RemoveAll(targetDir)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("unable to delete directory '%s' : %v", targetDir, err)
+	}
+
+	sourceFolderStat, statError := os.Stat(nc.algodDataDir)
+	if statError != nil {
+		err = statError
+		return
+	}
+
+	err = os.Mkdir(targetDir, sourceFolderStat.Mode())
```

Comment:
Here is a good idea to capture the error into a new variable, and copy it to err only if the error is not IsExist.
Otherwise, os.IsExist error may be returned.

Reply:
good idea.